capa/features/extractors/base_extractor.py

```diff
@@ -104,6 +130,12 @@ def get_base_address(self) -> Union[AbsoluteVirtualAddress, capa.features.addres
         """
         raise NotImplementedError()
 
+    def get_sample_hashes(self) -> SampleHashes:
+        """
+        fetch the hashes for the sample contained within the extractor.
+        """
+        return self.sample_hashes
```

Comment:
I think I'll move forward with the first approach, since getting the sample's hashes using ida (and binja's -- in the future) api is more elegant. There is an issue of repetition across extractors, but I think it's not that severe.